File: php/www/experiment/blog/quiz/testing/GetSubseriesTest.php
<?php
// GetSubseriesTest.php

assert_options(ASSERT_CALLBACK, function ($script, $line, $expression, $message) {
    printf("(%d) %s\n", $line, $message);
    exit;
});


require "getSubseries.php";

$series = [100,300,100,50,50,50,50,50,500,200,100];
$threshold = 500;

$result = getSubseries($series, $threshold);
assert(is_array($result), "Should return an array");

// this test passed initially, so was not the "next" test that should have been made
$sum = array_sum($result);
assert($sum <= $threshold, "Total of array elements ($sum) should be no more than the threshold ($threshold)");

// this is a valid second test
$length = count($result);
assert($length > 0, "For a threshold which should return values, the subseries should contain elements");

File: languageComparison/getSubseries/testing/python/getSubseriesTest.py
# getSubseriesTest.py

import json

from tests import getTests

#from daveWhite import subSeries as getSubseries
#from daveWhiteUpdated import getSubseries as getSubseries
from getSubseries import getSubseries

tests = getTests()


for test in tests:
	result = getSubseries(test['series'], test['threshold'])
	print(test['message'])
	if result == test['expected']:
		print("expected OK")
	else:
		print("*** expected MISMATCH. Expected: " + json.dumps(test['expected']) + "; result: " + json.dumps(result))

	if 'test' in test:
		if test['test'](result, test['threshold']):
			print("test OK")
		else:
			print("*** test FAILED")
	print("\n================================\n\n")

